include/net/aws_iot.h

```diff
@@ -28,7 +28,8 @@ extern "C" {
  *         topic that will be published to.
  */
 enum aws_iot_topic_type {
-	AWS_IOT_SHADOW_TOPIC_GET = 0x1,
+	AWS_IOT_SHADOW_TOPIC_UNKNOWN = 0x0,
```

Comment:
This "solution" is a bit lazy, I'll admit. All possible topics could be added to the enum, and topics on received messages could be parsed and assigned a more proper enum. What complicates it a bit is the option to set custom topics (non-standard AWS ones), which would not fit into any of the enums.